### ISSUE TITLE
[Rust] Non-`mut` locals are writable 

### ISSUE THREAD

**Issue by btj** [verifast/verifast#630]
The following program is accepted by VeriFast, even though it has Undefined Behavior:
```rust
unsafe fn mutate(p: *const i32)
//@ req *p |-> _;
//@ ens *p |-> 42;
{
    *(p as *mut i32) = 42;
}

fn foo()
//@ req true;
//@ ens true;
{
    let mut z = 24;
    unsafe { mutate(&raw const z); } // No UB

    let x = 24;
    unsafe { mutate(&raw const x); } // UB!
}
```

**Comment by btj** [comment]
This is somewhat tricky to handle since in the MIR, locals are not initialized right away. In the MIR, syntactically, initializing assignments and mutations look the same. The MIR for `foo` is as follows:
```
fn foo() -> () {
    let mut _0: ();
    let mut _1: i32;
    let _2: ();
    let mut _3: *const i32;
    let _5: ();
    let mut _6: *const i32;
    scope 1 {
        debug z => _1;
        let _4: i32;
        scope 2 {
            debug x => _4;
        }
    }

    bb0: {
        _1 = const 24_i32;
        _3 = &raw const _1;
        _2 = mutate(move _3) -> [return: bb1, unwind continue];
    }

    bb1: {
        _4 = const 24_i32;
        _6 = &raw const _4;
        _5 = mutate(move _6) -> [return: bb2, unwind continue];
    }

    bb2: {
        return;
    }
}
```

**Comment by btj** [comment]
Fixed by #635
